BrowserKit/Tests/SiteImageViewTests/ImageProcessing/SiteImageFetcherTests.swift

```diff
@@ -20,36 +20,33 @@ final class SiteImageFetcherTests: XCTestCase {
         self.mockImageDownloader = nil
     }
 
-    func testReturnsFailure_onAnyError() {
+    func testReturnsFailure_onAnyError() async {
         mockImageDownloader.error = KingfisherError.requestError(reason: .emptyRequest)
         let subject = DefaultSiteImageFetcher(imageDownloader: mockImageDownloader)
 
-        subject.fetchImage(imageURL: URL(string: "www.mozilla.com")!,
-                           completion: { result in
-            switch result {
-            case .success:
-                XCTFail("Should have failed with error")
-            case .failure(let error):
-                XCTAssertEqual("Unable to download image with reason: The request is empty or `nil`.",
-                               error.description)
-            }
-        })
+        let result = await subject.fetchImage(from: URL(string: "www.mozilla.com")!)
+
+        switch result {
+        case .success:
+            XCTFail("Should have failed with error")
+        case .failure(let error):
+            XCTAssertEqual("Unable to download image with reason: The request is empty or `nil`.",
+                           error.description)
+        }
```

Comment:
tests would actually check for a thrown error, or an actual success value, rather than switching on a result.